cmake/OpenCVDetectCXXCompiler.cmake

```diff
@@ -194,54 +194,7 @@ if(CMAKE_VERSION VERSION_LESS "3.1")
   endforeach()
 endif()
 
-if(NOT OPENCV_SKIP_CMAKE_CXX_STANDARD)
-  ocv_update(CMAKE_CXX_STANDARD 11)
```

Comment:
Nope.

Existed code allows compilation even with C++20.
Your patch breaks that and forces C++14.